code/network/lib/Loot/Network/ZMQ/Server.hs

```diff
@@ -109,7 +115,9 @@ createNetServEnv (ZTGlobalEnv ctx ztLogging) ztOurNodeId = liftIO $ do
 -- | Terminates server environment.
 termNetServEnv :: MonadIO m => ZTNetServEnv -> m ()
 termNetServEnv ZTNetServEnv{..} = liftIO $ do
+    adapterRelease ztServFrontAdapter
     Z.close ztServFront
+    adapterRelease ztServPubAdapter
```

Comment:
It can receive async exception even under `finally` since `A.cancel` falls into interruptible sleep